### PR TITLE
v1.3.0

### PR DESCRIPTION
**Pull request by TheTrunk**
General
- Automatically adjusts external IP address
- Do not log normal behavior of dropping ws connections
- UI improvements

ZelShare
- Rename of file/folder
- Show statistics of folder
- Download of folders
- Sharing of folders
- Shows download speeds
- Shows overall upload statistics (speed, data, progress)

Kadena
- It is now possible to set Kadena account nad Chain ID in ZelAdmin menu
- Posiblity to get Kadena account added

Log changes
- logs of warnings, info and debug are now being written to corresponding files
- Warn, info and debug logs are now available over api with both full download and last 100 lines

Following new API endpoints added
```
GET /zelnode/zelfluxwarnlog
GET /zelnode/zelfluxdebuglog
GET /zelnode/zelfluxinfolog
GET /zelnode/tailzelfluxerrorlog
GET /zelnode/tailzelfluxwarnlog
GET /zelnode/tailzelfluxdebuglog
GET /zelnode/tailzelfluxinfolog
GET /zelapps/zelshare/rename/:oldpath?/:newname?
GET /zelapps/zelshare/downloadfolder/:folder?
GET /zelflux/adjustkadena/:account?/:chainid?
GET /zelflux/kadena
```
Following API endpoints modified
```
/zelapps/zelshare/sharefile/:file? - can now point to a folder which would share a zip of folder
/zelapps/zelshare/unsharefile/:file? - can now point to a folder which would unshare a zip of folder
/zelid/adjustcruxid/:cruxid? -> /zeflux/adjustcruxid/:cruxid?
```

*Note* 
*Release not finalised. Expected release date 21st October. Expected features: KDA account and chainID add possibility*

### CODE REVIEW

**Review by Cabecinha84 — APPROVED** [review]
Ack.
Another great job.
Only concern - size of logs files. There is no rotation implemented I could check on my mobile review :)!

**Comment by TheTrunk** [comment]
> Only concern - size of logs files. There is no rotation implemented I could check on my mobile review :)!

Yes there is :) Log files are capped to 25MB. Then the log file will be overwritten.

File: ZelBack/src/lib/log.js [hunk 1]
```diff
@@ -18,41 +18,75 @@ function ensureString(parameter) {
   return typeof parameter === 'string' ? parameter : JSON.stringify(parameter);
 }
 
+function writeToFile(filepath, args) {
+  const size = getFilesizeInBytes(filepath);
+  let flag = 'a+';
+  if (size > (25 * 1000 * 1000)) { // 25MB
```
Comment (BIOHAZARDIQ) [inline ZelBack/src/lib/log.js:24]: 25 * 1024 * 1024

Comment (Cabecinha84) [reply ZelBack/src/lib/log.js:24]: nice find! 👍

**Pull request merged**